agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -153,6 +153,9 @@ type DockerClient interface {
 	// should be provided for the request.
 	ListContainers(context.Context, bool, time.Duration) ListContainersResponse
 
+	// ListImages return the set of the images known to the Docker daemon
+	ListImages(context.Context, time.Duration) ListImagesResponse
```

Comment:
we should probably use docker sdk here instead of fsouza to avoid more merge conflicts :(

Reply:
will create separate branch and rebase on top moby branch